### DISCUSSION TITLE
Performance issue occurs when selecting and un-selecting 10000 items in WINUI ListView control

### DISCUSSION THREAD

**Discussion by Subashini2129** [microsoft/microsoft-ui-xaml#7448] [Q&A]
Hi Team,

I am facing performance problem when selecting and un-selecting 1 lakh items in WINUI ListView control. Please find the sample and time taken details below:

**Selecting 1 lakh item -**  6.09 sec

**Un-selecting 1 lakh item -** 1.7 min

**Adding 10,000 items using SelectedItems property -** 2.7 min

**Sample:** 
[ListView-performance-sample.zip](https://github.com/microsoft/microsoft-ui-xaml/files/6936687/ListView-performance-sample.zip)

Can you please suggest any solution to improvise the performance issue?

Regards,
Subashini M.

**Comment by JPTGamesAndApps** [comment]
SelectedItems might be old school option from Windows 8. Try using SelectRange and see if it helps. I've noticed a significant performance improvement with that switch: https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.listviewbase.selectrange?view=winrt-20348

**Comment by StephenLPeters** [comment]
@Subashini2129 what version(s) of Winui are you using wheen you see this poor performance? What version of Windows are you using?

**Comment by Subashini2129** [comment]
@StephenLPeters please find the versions details below:

**WINUI version:**
NuGet package version:
[WinUI 3 - 0.8.1]

**Windows version**
Edition: Windows 10 Pro
Version: 20H2
OS build: 19042.1110

**Comment by Subashini2129** [comment]
@JPTGamesAndApps, I have used `DeselectRange` method of ListView to un-select 1 lakh item but still I am facing same performance delay of nearly 2 mins.

**Comment by SubashiniMahendran** [comment]
Hi Team,

Any update on this issue?
